### DISCUSSION TITLE
PassFields are not displaying on generated pass

### DISCUSSION THREAD

**Discussion by mbappai** [alexandercerutti/passkit-generator#130] [Q&A] [answered]
All fields (headers, primary, auxiliary) I specified don't seem to be showing up on the pass. Any reason as to why that is?

Here is how I'm implementing it

```
              ...
          
      export default async function handler(req:any, res:NextApiResponse){

    const body = JSON.parse(req?.body)

    const {qrPayload,expiryDate, location} = body

    const {signerCert, signerKey, wwdr, signerKeyPassphrase} = await getCertificates()
    const pass = await PKPass.from({
        model: path.resolve(
            __dirname,
            "../../../../",
            'pages/api/passkit/models/ticket.pass',
        ),
        certificates:{ 
            wwdr,
            signerKey,
            signerCert,
            signerKeyPassphrase
        }
    })

    // pass.set
    pass.headerFields.push(
        {
            key: "header1",
            label: "Event",
            value: "Flexable natural line skips",
            textAlignment: "PKTextAlignmentCenter",
        },
        {
            key: "header2",
            label: "Description",
            value: "Best service in Syracuse",
            textAlignment: "PKTextAlignmentCenter",
        },
    );

    pass.setLocations(location)

    pass.setBarcodes({
        message: `${JSON.stringify(qrPayload)}`,
        format: "PKBarcodeFormatQR",
        altText: "Qr code for bars"
    })

    

    pass.primaryFields.push({
        key: "Name",
        label: "Event Name",
        value: "Flexable natural line skips",
        textAlignment: "PKTextAlignmentLeft",
    })

    pass.setExpirationDate(new Date(expiryDate))

    pass.type = "eventTicket"
    
    const buffer = pass.getAsBuffer();  
    
    // console.log('streaming...',buffer)
    res.status(200)
    res.setHeader("Content-Type", pass.mimeType)
    // res.setHeader("Content-Dispostion", 'attachment; filena.pkpass')
    res.send(buffer)

    }  
    ....
```

PS: I have also set default values of those same field in pass.json

**Comment by alexandercerutti** [comment, marked as answer]
@mbappai you might want to check console logs or Console.app logs to check if there's some meaningful message. For sure, if you could share your code, it might let me help you further!

**Reply by mbappai** [reply]
Yeah, sorry about that just updated the question

**Reply by alexandercerutti** [reply]
I actually don't see anything wrong with your code 🤔
Does `Console.app` report anything?

**Reply by mbappai** [reply]
Nothing negative. Here is a screenshot
<img width="1440" alt="Screenshot 2023-03-17 at 10 23 59" src="https://user-images.githubusercontent.com/26521423/225864839-cd725d09-8257-47db-80b2-435717f59b14.png">

**Reply by mbappai** [reply]
I should however mention that the "relevant text" under QR code and image assets shows up. Here is a screenshot to that as well

<img alt="IMG_5B82B95C8B74-1" src="https://user-images.githubusercontent.com/26521423/225866694-d8332fec-8d1a-4baa-92f9-9c9d904392ab.jpeg" width="300">

**Reply by alexandercerutti** [reply]
Uhm, did you verify if `backgroundColor`, `foregroundColor` and `labelColor` have the same value perhaps? I don't know how your `pass.json` is formed, so I can only ipotesize some theories

**Reply by mbappai** [reply]
Yeah I previously didn't have labelColor set until you mentioned. It's still not showing up after giving it a color

**Reply by alexandercerutti** [reply]
Oh! I understood the issue: you are setting `pass.type` **after** having set the fields.

[As per the documention](https://github.com/alexandercerutti/passkit-generator/wiki/API-Documentation-Reference#:~:text=please%20note%20that%20they%20are%20strictly%20and%20directly%20linked%20to%20the%20pass%20type%20property%20(boardingpass%2C%20storecard%2C%20etc...).%20this%20means%20that%20if%20pass%20type%20is%20not%20available%2C%20accessing%20them%20will%20throw%20an%20error.%20changing%20the%20type%20on%20runtime%20will%20clean%20all%20them%20up.), fields are strictly related to `type`, so if you change it, they will get deleted.

**Reply by mbappai** [reply]
No shit!! haha It works now! 

Sorry I missed that part of the documentation. And besides i'm just realising that I already had the pass type set in the JSON so having it set dynamically is just redundant.

Thanks for the assistance as always Alex!

**Reply by alexandercerutti** [reply]
Just a tip for the future to debug: you can rename a `.pkpass` file into `.zip` and extract it and see what got generated 😉

**Reply by mbappai** [reply]
haha that is exactly what I did just now. Thanks for the tip though. Much appreciated!
